src/panic.md

```diff
@@ -10,56 +10,104 @@ Some language constructs, such as out-of-bounds [array indexing], panic automati
 
 r[panic.strategy]
 ## Panic strategy
 
 r[panic.strategy.intro]
-The _panic strategy_ defines the kind of panic runtime that a crate is built to support.
+The _panic strategy_ defines the kind of panic behavior that a crate is built to support.
 
 > [!NOTE]
 > The panic strategy can be chosen in `rustc` with the [`-C panic`] CLI flag.
+>
+> When generating a binary, dylib, cdylib, or staticlib and linking with `std`, the `-C panic` CLI flag also influences which [panic handler] is used.
 
 > [!NOTE]
 > When compiling code with the `abort` panic strategy, the optimizer may assume that unwinding across Rust frames is impossible, which can result in both code-size and runtime speed improvements.
 
 > [!NOTE]
-> See [link.unwinding] for restrictions on linking crates with different panic strategies. An implication is that crates built with the `unwind` strategy can use the `abort` runtime, but not vice-versa.
+> See [link.unwinding] for restrictions on linking crates with different panic strategies. An implication is that crates built with the `unwind` strategy can use the `abort` panic handler, but the `abort` strategy cannot use the `unwind` panic handler.
 
 r[panic.unwind]
 ## Unwinding
 
 r[panic.unwind.intro]
-Panicking may either be recoverable or non-recoverable, though it can be configured (by choosing the `abort` panic runtime) to always be non-recoverable. (The converse is not true: the `unwind` runtime does not guarantee that all panics are recoverable, only that panicking via the `panic!` macro and similar standard library mechanisms is recoverable.)
+Panicking may either be recoverable or non-recoverable, though it can be configured (by choosing a non-unwinding panic handler) to always be non-recoverable. (The converse is not true: the `unwind` handler does not guarantee that all panics are recoverable, only that panicking via the `panic!` macro and similar standard library mechanisms is recoverable.)
 
 r[panic.unwind.destruction]
-When panic recovery occurs, the `unwind` runtime "unwinds" Rust frames, just as C++'s `throw` unwinds C++ frames, until the panic reaches the point of recovery (for instance at a thread boundary). This means that as the panic traverses Rust frames, live objects in those frames that [implement `Drop`][destructors] will have their `drop` methods called. Thus, when normal execution resumes, no-longer-accessible objects will have been "cleaned up" just as if they had gone out of scope normally.
+When a panic occurs, the `unwind` handler "unwinds" Rust frames, just as C++'s `throw` unwinds C++ frames, until the panic reaches the point of recovery (for instance at a thread boundary). This means that as the panic traverses Rust frames, live objects in those frames that [implement `Drop`][destructors] will have their `drop` methods called. Thus, when normal execution resumes, no-longer-accessible objects will have been "cleaned up" just as if they had gone out of scope normally.
```

Comment:
The reason I specified "panic recovery" here is that, if there is no `catch_unwind` or thread boundary, then I _believe_ that a runtime using "two-phase unwinding" may not actually perform true "unwinding", and instead call `terminate()`: https://itanium-cxx-abi.github.io/cxx-abi/abi-eh.html#base-framework

...that said, this seems to imply that on most targets, C++ destructors don't actually get called before shutdown if there's a `throw` without a `catch`, and that seems....surprising.

In any case, since the second part of the sentence refers to "the point of recovery", maybe it's redundant to start with that here.